.github/workflows/publish-release.yml

```diff
@@ -45,10 +45,16 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # https://github.com/microsoft/vscode-ripgrep/issues/9
 
       - name: Publish NPM
-        shell: bash
-        run: |
-          yarn publish:latest -- ${{ inputs.release_type }}
-          yarn publish:check
+        # We need to retry the publish command in case of network errors.
+        # Those errors are usually due to rate limiting on the NPM registry.
+        uses: nick-invision/retry@7152eba30c6575329ac0576536151aca5a72780e # v3.0.0
+        with:
+          timeout_minutes: 5
+          retry_wait_seconds: 30
+          max_attempts: 3
+          retry_on: error
+          command: yarn publish:latest -- ${{ inputs.release_type }} && yarn publish:check
```

Comment:
The publishing guide recommends using a specific command in case of publishing errors:
https://github.com/eclipse-theia/theia/blob/master/doc/Publishing.md#failures-during-publishing
`npx lerna publish from-package --no-git-reset --no-git-tag-version --no-push`

I'm not familiar with the reasoning behind this choice. Maybe @vince-fugnitto could provide some insight?

Reply:
@jfaltermeier in the past we often encountered NPM socket errors which meant that only a subset of extensions were successfully published. In this state we cannot re-perform the publishing as it would fail with a message that the extension was already published. In order to only publish the extensions which were not previously published we manually ran the steps referenced in [failure-during-publishing](https://github.com/eclipse-theia/theia/blob/master/doc/Publishing.md#failures-during-publishing) to continue with the release.

It's possible that the proposed workflow will fail when retrying if we are in a state where we partially published extensions.

Reply:
Since we run `git reset --hard` in between retries, running `lerna publish from-package` is not necessary. Just repeating the `publish:latest` command should work as expected. I don't think we need to adjust something.

Also, having two separate commands in the CI (for normal publishing + publishing following an error) would make this more complicated to maintain imo.